plugins/woocommerce/src/Admin/RemoteInboxNotifications/RemoteInboxNotificationsEngine.php

```diff
@@ -105,6 +126,15 @@ public static function on_init() {
 		StoredStateSetupForProducts::init();
 	}
 
+	/**
+	 * Call the run function if the onboarding wizard has been completed.
+	 */
+	public static function possibly_run() {
+		if ( ! OnboardingProfile::needs_completion() || get_option( 'woocommerce_default_country', 'US:CA' ) !== 'US:CA' || ! empty( get_option( 'woocommerce_store_address', '' ) ) ) {
```

Comment:
I added the extra condition of the default_country check and store_address, as I am not sure if this would break things on existing stores.
Most if not all existing stores should have the onboarding wizard set to completed, but I don't know if there could be a way around that.

One other solution is to allow notes to be updated back to `pending` after being set to `unactioned`, we currently don't allow this, but this would fix the problem as well.
I decided to not go this route, as it could potentially cause strange behaviour as well, it might actually remove a lot of notes that rely on the store age. As a lot of those are time boxed between 1-7days for example, but maybe that is acceptable if the note was still `unactioned`?

Reply:
> I decided to not go this route, as it could potentially cause strange behaviour as well, it might actually remove a lot of notes that rely on the store age. As a lot of those are time boxed between 1-7days for example, but maybe that is acceptable if the note was still unactioned?

While I'm also not sure if it's safe, I think it makes sense to update `unactioned` notes back to `pending` when their rules no longer pass. 

cc @pmcpinto

Reply:
> I decided to not go this route, as it could potentially cause strange behaviour as well, it might actually remove a lot of notes that rely on the store age. As a lot of those are time boxed between 1-7days for example, but maybe that is acceptable if the note was still unactioned?


would this option remove notifications to any store or just existing stores?

Reply:
> It would remove notifications for any store only for the remote notifications. IF the conditions don't meet anymore and if the user hasn't clicked the note already.
For example the WooCommerce Shipping & Tax note, has a condition to show up for stores less than two day's old. This would mean it would be removed after the store is older then 2 days.

Mmm I think this can be problematic, as users would have a short window of time to interact with the notification.

Reply:
> Mmm I think this can be problematic, as users would have a short window of time to interact with the notification.

Yeah that is my thinking as well, I think [Moon's suggestion](https://github.com/woocommerce/woocommerce/pull/34632#pullrequestreview-1104454767) for this would be better, and allow things to work the same otherwise.